allauth/socialaccount/providers/openstreetmap/provider.py

```diff
@@ -16,10 +16,11 @@ def get_avatar_url(self):
         return self.account.extra_data.get("avatar")
 
     def get_username(self):
+        # TODO: this actually returns the display name, not the unique username
```

Comment:
I don't think there is anything that comes closer, given this example payload:

``` json
{
 "version": "0.6",
 "generator": "OpenStreetMap server",
 "user": {"id": 12023, "display_name": "jbpbis", "account_created": "2007-08-16T01:35:56Z", "description": "", "contributor_terms": {"agreed": False}, "roles": [], "changesets": {"count": 1}, "traces": {"count": 0}, "blocks": {"received": {"count": 0, "active": 0}}}
}
```

See: https://wiki.openstreetmap.org/wiki/API_v0.6